src/components/Content.jsx

```diff
@@ -0,0 +1,45 @@
+import { Route, Routes } from 'react-router-dom';
+import { ArticleList } from '../pages/ArticleList';
+import { MostRecentArticles } from '../pages/MostRecentArticles';
+import { useEffect, useState } from 'react';
+import { getAllArticles } from '../utils/api';
+
+export const Content = () => {
+  const [articleList, setArticleList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
```

Comment:
The 'isLoading' state could be set to true here just before the API request, so then it would indicate loading even later when you want to fetch the articles again on the page.